### PR TITLE
Use external libs

### PR DESCRIPTION
**Pull request by wolfv**
These are changes ported from `vcpkg` (which in turn looks like it ported changes from Fedora RPMs to unbundle the underlying libraries.

This is necessary in order to e.g. link with QT and freeimage at the same time.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,49 +1,62 @@
...
+        - Makefile.osx.patch  # [osx] 
 
 build:
-    number: 0
-    skip: true  # [win and vc<14]
-    binary_relocation: False  # [osx]
-    run_exports:
-        - {{ pin_subpackage('freeimage', max_pin='x.x') }}
+  number: 1
+  skip: true  # [win or osx]
```
Comment (isuruf) [inline recipe/meta.yaml]: Can you enable OSX? I don't think we can merge with OSX disabled.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,49 +1,62 @@
...
+    - libwebp
 
 test:
-    requires:
-        - python
-    commands:
-        - python run_test.py
+  commands:
+    - test -f $PREFIX/lib/libfreeimage${SHLIB_EXT}  # [unix]
+    - test -f $PREFIX/lib/libfreeimage-3.18.0${SHLIB_EXT}  # [unix]
```
Comment (isuruf) [inline recipe/meta.yaml]: Use `version` instead of 3.18.0 ?

File: recipe/meta.yaml (hunk 1 above)
Comment (wolfv) [reply recipe/meta.yaml]: why can we not?
I had issues with libpng and I currently don't have a OS X to test locally. (I think the patch you pointed me at will not fix it).
But I'll attempt a build let's see.

File: recipe/meta.yaml (hunk 2 above)
Comment (wolfv) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml (hunk 1 above)
Comment (wolfv) [reply recipe/meta.yaml]: ok doesn't build . 

can we merge like this for now until i have a mac mini?

Comment (isuruf) [reply recipe/meta.yaml]: cc @conda-forge/freeimage

Comment (beckermr) [reply recipe/meta.yaml]: You can use the azure to test builds. We have enough capacity that submitting commits to test things is totally fine. You don't need your own mac to test osx.

**Comment by wolfv** [comment]
please @isuruf or @beckermr merge this, I can't waste more time on this stuff right now. the cmake based solution would have worked fine by now and I don't want to further debug OS X on azure it's really tedious. 

once i have a local machine i can make a proper build for OS X.

**Comment by wolfv** [comment]
@jakirkham @conda-forge/freeimage could someone take a look and merge please? this is blocking gazebo11 over here: github.com/conda-forge/gazebo-feedstock

**Comment by wolfv** [comment]
@isuruf OS X is done now as well.

**Pull request closed without merging**